src/main/java/juja/microservices/teams/service/TeamService.java

```diff
@@ -3,54 +3,84 @@
-        logger.info("Added new Team with parameters '{}'", team);
-        logger.debug("Finish TeamService.addTeam. newTeamId: {}", newTeamId);
-        return newTeamId;
+        log.debug("Started 'Save team '{}'", team);
+        teamRepository.saveTeam(team);
+        log.info("Finished 'Save team' '{}'", team.getId());
+        log.debug("Finished 'Save team '{}'", team);
+        return team;
     }
 
     private Team mappingRequestToTeam(TeamRequest teamRequest) {
-        return new Team(teamRequest.getFrom(), teamRequest.getUuidOne(), teamRequest.getUuidTwo(),
-                teamRequest.getUuidThree(), teamRequest.getUuidFour());
+        return new Team(teamRequest.getMembers());
     }
 
-    private boolean isAnyBodyInOtherTeam(TeamRequest team){
-        logger.debug("Start isAnyBodyInOtherTeam()");
-        boolean result = false;
-        if (teamRepository.isUserInOtherTeam(team.getUuidOne()) ||
-                teamRepository.isUserInOtherTeam(team.getUuidTwo()) ||
-                teamRepository.isUserInOtherTeam(team.getUuidThree()) ||
-                teamRepository.isUserInOtherTeam(team.getUuidFour())) {
-            result = true;
+    private Set<String> usersInCurrentTeams(TeamRequest teamRequest) {
+        log.debug("Started 'usersInCurrentTeams' with teamRequest '{}'",teamRequest);
+        Set<String> usersInTeams = new HashSet<>();
+        teamRequest.getMembers()
+                .forEach(uuid -> {
+                    List<Team> teams = teamRepository.getUserTeams(uuid);
+                    if (teams.size() != 0) {
+                        usersInTeams.add(uuid);
+                    }
+                });
+        log.info("Finished 'usersInCurrentTeams' with teamRequest '{}'",teamRequest);
+        return usersInTeams;
+    }
+
+    public Team deactivateTeam(String uuid) {
+        log.debug("Started 'deactivateTeam' with uuid '{}'",uuid);
+        List<Team> teams = getUserTeams(uuid);
+        log.info("Finished 'usersInCurrentTeams' with uuid '{}. Teams size '",uuid,teams.size());
+        log.debug("Finished 'usersInCurrentTeams' with uuid '{}'. Teams ",uuid,teams.toString());
+        if (teams.size() == 1) {
+            Team team = teams.get(0);
+            team.setDeactivateDate(LocalDate.now());
+            return teamRepository.saveTeam(team);
+        } else if (teams.size() == 0) {
+            log.warn("User <{}> is not in the team now", uuid);
+            throw new UserNotInTeamException(String.format("User with uuid '%s' not in team now", uuid));
+        } else {
+            log.warn("User <{}> is in several teams ", uuid);
+            throw new UserInSeveralTeamsException(String.format("User with uuid '%s' is in several teams now", uuid));
         }
-        logger.info("Result execution isAnyBodyInOtherTeam is {}", result);
-        logger.debug("Finish isAnyBodyInOtherTeam()");
-        return result;
+    }
+
+    private List<Team> getUserTeams(String uuid) {
```

Comment:
This method needed in the task TMF-F4 and F5. Get user group by uuid. Lately this method be refactored.